doc/source/whatsnew/v0.20.0.txt

```diff
@@ -705,6 +707,133 @@ New Behavior:
    TypeError: Cannot compare 2014-01-01 00:00:00 of
    type <class 'pandas.tslib.Timestamp'> to string column
 
+.. _whatsnew_0200.api_breaking.index_order:
+
+Index order after inner join due to Index intersection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``Index.intersection`` now preserves the order of the calling Index (left)
```

Comment:
say what this did previously. No need to mention ``Index.join`` or ``DataFrame.merge`` and say ``.align`` methods. simpler is better.